src/operator/fully_connected-inl.h

```diff
@@ -159,8 +159,10 @@ class FullyConnectedProp : public OperatorProperty {
     if (dshape.ndim() ==  0) return false;
 
     index_t num_input = 0;
-    mshadow::Shape<2> ishape = dshape.FlatTo2D();
-    num_input = ishape[1];
+    num_input = 1;
+    for (int i = 1; i < dshape.ndim(); ++i)
```

Comment:
Can you add a PR to add ProdShape to TShape? I guess this could be frequently used in other places